pkg/onchain/sc.go

```diff
@@ -111,10 +111,23 @@ func DeploySC(client *node.Client, wallet wallet.Wallet, contract []byte) (strin
 			return "", fmt.Errorf("waiting SC deployment: %w", err)
 		}
 
+		event := events[0].Data
+		address := strings.Split(event, ":")[1]
+
 		if len(events) > 0 {
-			return strings.Split(events[0].Data, ":")[1], nil
+			//  Catch Run Time Error and return it
+			if strings.Contains(event, "massa_execution_error") {
+				return "", errors.New("runtime error")
+			}
+
+			// Catch Address with generic event parameter, and check if it's a valid address
+			if strings.Contains(event, eventParameter) {
```

Comment:
In fact, I think that we shouldn't check anything and, as long as an event is returned (and that there are no runtime errors), we return the event.

It is to the caller of the function to parse this event however he sees fit.
In our case, the website uploader would call this function and then parse the event on his side, looking for the address of the SC. This way I think it's generic enough.

WDYT ?

Reply:
I aggree i think it's a good idea. We shift the responsibility to the implementation.